src/Portability/Middleware.php

```diff
@@ -4,15 +4,18 @@
 
 namespace Doctrine\DBAL\Portability;
 
+use Doctrine\DBAL\ColumnCase;
 use Doctrine\DBAL\Driver as DriverInterface;
 use Doctrine\DBAL\Driver\Middleware as MiddlewareInterface;
 
 final class Middleware implements MiddlewareInterface
 {
     private int $mode;
 
+    /** @var 0|ColumnCase::LOWER|ColumnCase::UPPER */
     private int $case;
 
+    /** @param 0|ColumnCase::LOWER|ColumnCase::UPPER $case */
     public function __construct(int $mode, int $case)
```

Comment:
I think we can also be more precise about the `$mode` parameter, but that can be done in a follow-up.

Reply:
I've added a description. Please let me know if you think it is enough.